addon/record-data.js

```diff
@@ -27,18 +27,21 @@ export default class FragmentRecordData extends RecordData {
       super(identifier.type, identifier.id, identifier.clientId, store);
     }
 
+    // @patocallaghan - We need to keep a copy of the record so we can recreate the fragment/fragmentArray.
+    this._record = null;
```

Comment:
I couldn't find some "happy path" to retrieve a record from `RecordData` so I resort to setting this private field throughout the various hooks where the record is passed through.